extensions/amp-position-observer/0.1/amp-position-observer.js

```diff
@@ -359,8 +359,9 @@ export class AmpVisibilityObserver extends AMP.BaseElement {
     } else {
       scene = this.element.parentNode;
     }
-    // Hoist body to documentElement.
-    if (this.getAmpDoc().getBody() == scene) {
+    // Hoist body to documentElement, expected to work for both ampdoc and
+    // amp4ads in friendly iframe.
+    if (scene && scene.tagName === 'BODY') {
```

Comment:
might be better to check `this.win.document.body === scene`

Also L356 seems to be buggy.

I guess eventually we will need `ampdoc-fie` launch (/cc @dvoytenko) to completely address the issue. Please add a TODO.

Reply:
Yeah that would make sense.